pkg/frontend/admin_hive_syncset_resources.go

```diff
@@ -0,0 +1,65 @@
+package frontend
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"net/http"
+	"path/filepath"
+	"strings"
+
+	"github.com/sirupsen/logrus"
+	"github.com/ugorji/go/codec"
+
+	"github.com/Azure/ARO-RP/pkg/api"
+	"github.com/Azure/ARO-RP/pkg/frontend/middleware"
+)
+
+func (f *frontend) getAdminHiveClusterSync(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+	log := ctx.Value(middleware.ContextKeyLog).(*logrus.Entry)
+	resourceId := strings.TrimPrefix(filepath.Dir(r.URL.Path), "/admin")
+	b, err := f._getAdminHiveClusterSync(ctx, resourceId)
+
+	if cloudErr, ok := err.(*api.CloudError); ok {
+		api.WriteCloudError(w, cloudErr)
+		return
+	}
+
+	adminReply(log, w, nil, b, err)
+}
+
+func (f *frontend) _getAdminHiveClusterSync(ctx context.Context, resourceId string) ([]byte, error) {
+	// We have to check if the frontend has a valid clustermanager since hive is not everywhere.
+	if f.hiveClusterManager == nil {
+		return nil, api.NewCloudError(http.StatusBadRequest, api.CloudErrorCodeInvalidParameter, "", "hive is not enabled")
+	}
+
+	dbOpenShiftClusters, err := f.dbGroup.OpenShiftClusters()
+	if err != nil {
+		return nil, api.NewCloudError(http.StatusInternalServerError, api.CloudErrorCodeInternalServerError, "", err.Error())
+	}
+
+	doc, err := dbOpenShiftClusters.Get(ctx, resourceId)
+	if err != nil {
+		return nil, api.NewCloudError(http.StatusNotFound, api.CloudErrorCodeNotFound, "", err.Error())
```

Comment:
nit:
It can be something like this.
https://github.com/Azure/ARO-RP/blob/f1fbd0feb24f3b4a1dbba53bf0f9ceff0b51020d/pkg/frontend/admin_openshiftcluster_approvecsr.go#L55-L60

otherwise it returns 404 when for example network failure or something internal server error happens.
Since this is not user-facing and there are some other similar error handling, it's not a big deal though.

Reply:
We already know the resourceId, so we have these details in that. Is it mandatory to give these details explicitly in the error message?

Reply:
The problem is that it could returns 404 even if it exists when there's something else wrong.
This might give false impression that there's not such a cluster, but actually it exists.
We can avoid that situation by switch based on the cosmosdb error type.